VideoApi/VideoApi.UnitTests/Services/VirtualRoom/GetOrCreateAnInterpreterVirtualRoomTests.cs

```diff
@@ -0,0 +1,124 @@
+            
+
+            _mocker.Mock<IQueryHandler>().Setup(x =>
+                    x.Handle<GetAvailableRoomByRoomTypeQuery, List<Room>>(It.Is<GetAvailableRoomByRoomTypeQuery>(q =>
+                        q.ConferenceId == _conference.Id && q.CourtRoomType == VirtualCourtRoomType.Civilian)))
+                .ReturnsAsync(new List<Room> {emptyInterpreterRoom, interpreterRoom});
+
+            var room = await _service.GetOrCreateAnInterpreterVirtualRoom(_conference, participant);
+            room.Should().NotBeNull();
+            room.Should().BeEquivalentTo(interpreterRoom);
+        }
+
+        [Test]
+        public async Task should_create_vmr_with_kinly_if_room_is_not_available()
+        {
+            var participant = _conference.Participants.First(x => !x.IsJudge());
+            var expectedRoom = new Room(_conference.Id, VirtualCourtRoomType.Civilian, false);
+            var newVmrRoom = new BookedParticipantRoomResponse
+            {
+                Display_name = "InterpreterRoom1",
+                Uris = new Uris
+                {
+                    Participant = "wertyui__interpreter",
+                    Pexip_node = "test.node.com"
+                }
+            };
+            
+            
+            _mocker.Mock<IQueryHandler>().SetupSequence(x =>
+                    x.Handle<GetAvailableRoomByRoomTypeQuery, List<Room>>(It.Is<GetAvailableRoomByRoomTypeQuery>(q =>
+                        q.ConferenceId == _conference.Id && q.CourtRoomType == VirtualCourtRoomType.Civilian)))
+                .ReturnsAsync(new List<Room>())
+                .ReturnsAsync(new List<Room>{expectedRoom});
+
+            _mocker.Mock<ICommandHandler>().Setup(x =>
+                x.Handle(It.IsAny<UpdateRoomCommand>())).Callback(() =>
+                expectedRoom.UpdateRoomDetails(newVmrRoom.Display_name, "ingesturl", newVmrRoom.Uris.Pexip_node,
+                    newVmrRoom.Uris.Participant));
+
+            _mocker.Mock<IKinlyApiClient>().Setup(x =>
+                x.CreateParticipantRoomAsync(_conference.Id.ToString(), It.IsAny<CreateParticipantRoomParams>()))
+                .ReturnsAsync(newVmrRoom);
+            
+            var room = await _service.GetOrCreateAnInterpreterVirtualRoom(_conference, participant);
+
+            room.Should().NotBeNull();
+            room.Label.Should().Be(newVmrRoom.Display_name);
+            room.PexipNode.Should().Be(newVmrRoom.Uris.Pexip_node);
+            room.ParticipantUri.Should().Be(newVmrRoom.Uris.Participant);
+        }
+
+        private Conference InitConference()
+        {
+            var conference = new ConferenceBuilder().WithParticipants(3).Build();
+            var participantA = conference.Participants[0];
+            var participantB = conference.Participants[1];
+            
+            participantA.AddLink(participantB.Id, LinkedParticipantType.Interpreter);
+            participantB.AddLink(participantA.Id, LinkedParticipantType.Interpreter);
```

Comment:
wrong link type

Reply:
There's only one type of link enum. The interpreter type was dropped since we use the hearing role to determine who is who